app/helpers/courses_helper.rb

```diff
@@ -49,4 +49,14 @@ def percent_complete(course)
     end
     ""
   end
+
+  def percent_complete_without_user(course, lesson_id)
+    session[:completed_lessons] ||= []
+    total_lessons = course.lessons.count
+    session[:completed_lessons] << lesson_id unless session[:completed_lessons].include?(lesson_id)
+    completed = session[:completed_lessons].count
+    return 0 if total_lessons == 0
```

Comment:
@coreyd303 What if this was done like:

```
percent_complete(completed, total_lessons)
```
```
def percent_complete(completed_count, total_lessons)
  if total_lessons == 0
    0
  else
   ((completed_count.to_f / total_lessons) * 100).round
  end
end
```

That way, you'd remove the explicit return (always good to have less of those), refactor out the logic for calculating the percent into a reusable function you could easily test and use elsewhere, and overall make things a little easier to read through.